src/node_resource_limits.cc

```diff
@@ -0,0 +1,72 @@
+#include "node_resource_limits.h"
+
+#include <fstream>
+
+#ifdef __POSIX__
+#include <sys/stat.h>
+#endif  // __POSIX__
+
+namespace node {
+namespace resource_limits {
+
+#ifdef __POSIX__
+bool CgroupsResourceLimits::CgroupsAvailable(std::string cgroups_directory) {
+  struct stat stat_buf;
+  int err = stat(cgroups_directory.c_str(), &stat_buf);
+  if (err) {
+    return false;
+  } else {
+    return !!(stat_buf.st_mode & S_IFDIR);
+  }
+}
+
+bool CgroupsResourceLimits::MemParse(uint64_t* output, std::string input) {
```

Comment:
style nit: it is conventional for the out parameter to come after the in parameters.